internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -161,10 +158,18 @@ func (r *AgentServiceConfigReconciler) Reconcile(origCtx context.Context, req ct
 	log.Info("AgentServiceConfig Reconcile started")
 
 	instance := &aiv1beta1.AgentServiceConfig{}
+	ascKey := types.NamespacedName{Name: req.NamespacedName.Name}
 
-	// NOTE: ignoring the Namespace that seems to get set on request when syncing on namespaced objects
-	// when our AgentServiceConfig is ClusterScoped.
-	if err := r.Get(ctx, types.NamespacedName{Name: req.NamespacedName.Name}, instance); err != nil {
+	//for backwards compatability, if the namespace is assisted-installer we are looking
+	//for a cluster scoped CRD. Otherwise, we will look for a namespace-scoped resource
+	//and implicitly assume that we are in hypershift mode (L0-L1)
+	if req.NamespacedName.Namespace != "" {
```

Comment:
When would this be empty?

Reply:
It is empty now (with the cluster scoped version). All other resources however do have the namespace "assisted-installer"

Reply:
Right, but `req.NamespacedName` will always be the key for an `AgentServiceConfig` instance.
And after this PR that should always have a namespace, right?

Reply:
I think this ties into what happens on upgrade.